filters/include/pcl/filters/impl/extract_indices.hpp

```diff
@@ -147,12 +147,12 @@ pcl::ExtractIndices<PointT>::applyFilterIndices (std::vector<int> &indices)
   else  // Inverted functionality
   {
     // Set up the full indices set
-    std::vector<int> full_indices (input_->size ());
-    for (int fii = 0; fii < static_cast<int> (full_indices.size ()); ++fii)  // fii = full indices iterator
+    Indices full_indices (input_->size ());
+    for (index_t fii = 0; fii < static_cast<index_t> (full_indices.size ()); ++fii)  // fii = full indices iterator
```

Comment:
`std::iota`? (and in L135)

Reply:
I wonder if I write `std::iota(full_indices.begin(), full_indices.end(), 0)`, what type will the 0 be? My guess is `int`. So would `std::iota(full_indices.begin(), full_indices.end(), static_cast<index_t>(0))` be sufficient?

Reply:
Yeah. The 2nd form is needed

Reply:
Done